odoo/addons/base/models/res_partner.py

```diff
@@ -239,6 +239,42 @@ def default_get(self, default_fields):
         ('check_name', "CHECK( (type='contact' AND name IS NOT NULL) or (type!='contact') )", 'Contacts require a name'),
     ]
 
+    @api.depends('name', 'user_ids', 'image_1920')
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends('name', 'user_ids', 'image_1024')
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends('name', 'user_ids', 'image_512')
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends('name', 'user_ids', 'image_256')
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends('name', 'user_ids', 'image_128')
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _get_one_avatar(self, avatar_field, image_field):
+        self.ensure_one()
+        if self.user_ids - self.user_ids.filtered('share'):
+            return super()._generate_avatar_svg(self[self._rec_name])
+        return self._get_avatar_placeholder()
+
+    def _get_avatar_placeholder(self):
+        image = 'company_image.png' if self.is_company else 'avatar_grey.png'
+        image_path = get_resource_path('base', 'static/img', image)
+        return base64.b64encode(tools.file_open(image_path, 'rb').read())
```

Comment:
Follow up comment from MAT, if we can call `file_open` with the full path, I would do it here too. Avoid building the path dynamically.